specs/ExecuteScript.md

```diff
@@ -0,0 +1,222 @@
+  /// navigations or `history.pushState` navigations).  This operation
+  /// follows the `NavigationStarting` and precedes `NavigationCompleted` events.
+  HRESULT add_ContentLoading(
+      [in] ICoreWebView2FrameContentLoadingEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_ContentLoading`.
+  HRESULT remove_ContentLoading(
+      [in] EventRegistrationToken token);
+
+  /// Add an event handler for the `NavigationCompleted` event.
+  /// `NavigationCompleted` runs when the CoreWebView2Frame has completely
+  /// loaded (`body.onload` runs) or loading stopped with error.
+  ///
+  /// \snippet ScenarioWebViewEventMonitor.cpp NavigationCompleted
+  HRESULT add_NavigationCompleted(
+      [in] ICoreWebView2FrameNavigationCompletedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+
+  /// Remove an event handler previously added with `add_NavigationCompleted`.
+  HRESULT remove_NavigationCompleted(
+      [in] EventRegistrationToken token);
+
+  /// Add an event handler for the DOMContentLoaded event.
+  /// DOMContentLoaded is raised when the iframe html document has been parsed.
+  /// This aligns with the the document's DOMContentLoaded event in html.
+  ///
+  /// \snippet ScenarioWebViewEventMonitor.cpp DOMContentLoaded
+  HRESULT add_DOMContentLoaded(
+      [in] ICoreWebView2FrameDOMContentLoadedEventHandler* eventHandler,
+      [out] EventRegistrationToken* token);
+  /// Remove an event handler previously added with add_DOMContentLoaded.
+  HRESULT remove_DOMContentLoaded(
+      [in] EventRegistrationToken token);
+
+  /// Run JavaScript code from the javascript parameter in the current frame.
+  /// The result of evaluating the provided JavaScript is used in this parameter.
+  /// The result value is a JSON encoded string. If the result is undefined,
+  /// contains a reference cycle, or otherwise is not able to be encoded into
+  /// JSON, the JSON `null` value is returned as the `null` string.
+  ///
+  /// \> [!NOTE]\n\> A function that has no explicit return value returns undefined.  If the
+  /// script that was run throws an unhandled exception, then the result is
+  /// also `null`. This method is applied asynchronously. If the method is
+  /// run before `ContentLoading`, the script will not be executed
+  /// and the JSON `null` will be returned.
+  /// This operation works even if `ICoreWebView2Settings::IsScriptEnabled` is
+  /// set to `FALSE`.
+  ///
+  /// \snippet ScriptComponent.cpp ExecuteScriptFrame
+  HRESULT ExecuteScript(
+      [in] LPCWSTR javaScript,
+      [in] ICoreWebView2ExecuteScriptCompletedHandler* handler);
+}
+```
+## .NET and WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+    /// CoreWebView2Frame provides direct access to the iframes information and use.
+     runtimeclass CoreWebView2Frame
+    {
+        /// A frame navigation will raise a `NavigationStarting` event and
```

Comment:
Process note ... OK to leave out the descriptions here that duplicate above. Nothing wrong with it, but if you want to save some time.

Reply:
👍